BitwardenShared/Core/Platform/Services/AuthenticatorSyncService.swift

```diff
@@ -174,22 +174,40 @@ actor DefaultAuthenticatorSyncService: NSObject, AuthenticatorSyncService {
         }
     }
 
+    /// If sync has been turned off for all accounts, delete the Authenticator key from the shared keychain.
+    ///
+    private func deleteKeyIfSyncingIsOff() async throws {
+        var hasAccountWithSync = false
+        for account in try await stateService.getAccounts() {
+            hasAccountWithSync = try await stateService.getSyncToAuthenticator(userId: account.profile.userId)
+            if hasAccountWithSync {
+                break
+            }
+        }
+        if !hasAccountWithSync {
+            try? sharedKeychainRepository.deleteAuthenticatorKey()
+        }
+    }
```

Comment:
🎨 You can simplify this a bit like this:
```suggestion
    private func deleteKeyIfSyncingIsOff() async throws {
        for account in try await stateService.getAccounts() {
            let hasAccountWithSync = try await stateService.getSyncToAuthenticator(userId: account.profile.userId)
            guard !hasAccountWithSync else {
                return
            }
        }

        try? sharedKeychainRepository.deleteAuthenticatorKey()
    }
```